File: test/e2e/test-files/workspaces/nyc-flights-data-py/flights-5million.py
import pandas as pd
import os

# Read the original file
data_file_path = os.path.join(os.getcwd(), 'data-files', 'flights', 'flights.parquet')
df = pd.read_parquet(data_file_path, engine='pyarrow')

# Repeat the DataFrame 15 times ~ 5 million rows
df_5mil = pd.concat([df] * 15, ignore_index=True)

# Verify the size
print(f"Number of rows in the expanded DataFrame: {len(df_5mil)}")

File: test/e2e/test-files/workspaces/nyc-flights-data-r/flights-5million.r
library(arrow)
library(dplyr)

# Build the file path (similar to os.path.join in Python)
data_file_path <- file.path(getwd(), "data-files", "flights", "flights.parquet")

# Read the original file
df <- read_parquet(data_file_path)

# Repeat the DataFrame 15 times ~ 5 million rows
df_5mil <- bind_rows(replicate(15, df, simplify = FALSE))

# Verify the size
cat("Number of rows in the expanded DataFrame:", nrow(df_5mil), "\n")